detox/src/devices/common/drivers/android/tools/ApkValidator.test.js

```diff
@@ -0,0 +1,54 @@
+describe('APK validation', () => {
+
+  const binaryPath = 'mock-bin-path';
+  const testBinaryPath = 'mock-test-bin-path';
+
+  let aapt;
+  let uut;
+  beforeEach(() => {
+    const AAPT = jest.genMockFromModule('../exec/AAPT');
+    aapt = new AAPT();
+
+    const ApkValidator = require('./ApkValidator');
+    uut = new ApkValidator(aapt);
+  });
+
+  const givenAAPTResult = (result) => aapt.isTestAPK.mockResolvedValue(result);
+  const givenAAPTError = (error) => aapt.isTestAPK.mockRejectedValue(error);
+
+  describe('App APK validation', () => {
+    it('should validate the APK is not a test APK', async () => {
+      givenAAPTResult(false);
+      await uut.validateAppApk(binaryPath);
+      expect(aapt.isTestAPK).toHaveBeenCalledWith(binaryPath);
+    });
+
+    it('should throw a descriptive error if app APK happens to be a test APK', async () => {
+      givenAAPTResult(true);
+      await expect(uut.validateAppApk(binaryPath)).rejects.toThrowErrorMatchingSnapshot();
+    });
```

Comment:
I'd add expectation in all tests (`expect(aapt.isTestAPK).toHaveBeenCalledWith(binaryPath);`)

Reply:
Good catch!

Reply:
Ohh, I though you meant something else. Simply adding as-is would be redundant :-/